lib/debug_timer/node.rb

```diff
@@ -1,22 +1,36 @@
 module DebugTimer
   class Node
-    attr_reader :name, :children, :start_time, :parent, :time_elapsed
-
+    attr_reader :name, :children, :start_time, :parent, :time_elapsed, :start_gc_stats, :end_gc_stats
     def initialize(name)
       @name = name
       @children = []
+      if DebugTimer.object_allocations?
+        grab_object_allocations(true)
+      end
       @start_time = Time.now
     end
 
     def stop
       @time_elapsed = Time.now - start_time
+      if DebugTimer.object_allocations?
+        grab_object_allocations(false)
```

Comment:
This feels kind of weird but I don't know of a better way to do it.  Suggestions?